.ci/pipelines/e2e-tests-testmutationsecondmastersetdown.Jenkinsfile

```diff
@@ -0,0 +1,112 @@
+// This library overrides the default checkout behavior to enable sleep+retries if there are errors
+// Added to help overcome some recurring github connection issues
+@Library('apm@current') _
+
+def failedTests = []
+def lib
+
+pipeline {
+
+    agent {
+        label 'linux'
+    }
+
+    options {
+        timeout(time: 300, unit: 'MINUTES')
+    }
+
+    environment {
+        VAULT_ADDR = credentials('vault-addr')
+        VAULT_ROLE_ID = credentials('vault-role-id')
+        VAULT_SECRET_ID = credentials('vault-secret-id')
+        GCLOUD_PROJECT = credentials('k8s-operators-gcloud-project')
+    }
+
+    stages {
+        stage('Checkout from GitHub') {
+            steps {
+                checkout scm
+            }
+        }
+        stage('Load common scripts') {
+            steps {
+                script {
+                    lib = load ".ci/common/tests.groovy"
+                }
+            }
+        }
+       /* stage('Run Checks') {
+            when {
+                expression {
+                    notOnlyDocs()
+                }
+            }
+            steps {
+                sh 'make -C .ci TARGET=ci-check ci'
+            }
+        }*/
+        stage("E2E tests") {
+            when {
+                expression {
+                    notOnlyDocs()
+                }
+            }
+            steps {
+                sh '.ci/setenvconfig e2e/master'
+                
+                sh 'echo TESTS_MATCH = TestMutationSecondMasterSetDown >> .env'
+                sh 'echo E2E_SKIP_CLEANUP = true >> .env'
+                sh 'sed "s/CLUSTER_NAME=eck-e2e/CLUSTER_NAME=eck-debug-e2e/" -i .env'
+                
+                script {
+                    env.SHELL_EXIT_CODE = sh(returnStatus: true, script: 'make -C .ci get-test-artifacts TARGET=ci-build-operator-e2e-run ci')
+
+                    //sh 'make -C .ci TARGET=e2e-generate-xml ci'
+                    //junit "e2e-tests.xml"
+
+                    if (env.SHELL_EXIT_CODE != 0) {
+                        failedTests = lib.getListOfFailedTests()
+                    }
+
+                    sh 'exit $SHELL_EXIT_CODE'
+                }
+            }
+        }
+    }
+
+    post {
+        unsuccessful {
+            script {
+                def msg = lib.generateSlackMessage("E2E tests failed!", env.BUILD_URL, failedTests)
+
+                slackSend(
+                      channel: '#cloud-k8s',
+                      color: 'danger',
+                      message: msg,
+                    tokenCredentialId: 'cloud-ci-slack-integration-token',
+                    botUser: true,
+                    failOnError: true
+                )
+            }
+        }
+        cleanup {
+            script {
+                if (notOnlyDocs() && failedTests.size == 0) {
```

Comment:
This does not work. I am not 100% sure why, but when we ran this clusters were never cleaned up. I deleted the ones for successful runs manually once we had exceeded our quota.

Reply:
In Jenkins, a scripted pipeline is a groovy-based DSL and in this context, we can't omit the parentheses when we call `size`:
```groovy
if (notOnlyDocs() && failedTests.size() == 0) {
```

Reply:
Also will `failedTests.size()` not always be 0 because we have commented out the junit bits that report failed test into Jenkins?

Reply:
it looks like you're probably right? We can try and update it if we end up needing it to debug in the future